platform/android/display_server_android.cpp

```diff
@@ -531,6 +531,11 @@ void DisplayServerAndroid::notify_surface_changed(int p_width, int p_height) {
 DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, DisplayServer::WindowMode p_mode, DisplayServer::VSyncMode p_vsync_mode, uint32_t p_flags, const Vector2i *p_position, const Vector2i &p_resolution, int p_screen, Error &r_error) {
 	rendering_driver = p_rendering_driver;
 
+	bool tts_enabled = GLOBAL_GET("audio/general/text_to_speech");
+	if (tts_enabled) {
+		TTS_Android::init();
+	}
+
```

Comment:
Shouldn't this be in `tts_android` or in the java layer itself?
While this is not the case now, I'm primarily concerned about any changes that may alter the init order of `TTS_Android` and `DisplayServerAndroid` and thus create a bug.

Reply:
`TTS_Android` is directly used only from display server, the rest of code calls it via display server wrappers. And `GodotTTS` on the Java side is created before project setting are available.

Reply:
`TTS_Android` is initialized in [`java_godot_lib_jni.cpp`](https://github.com/godotengine/godot/blob/437041a983a81f9768673dc5374f644634f940f9/platform/android/java_godot_lib_jni.cpp#L136) via [`TTS_Android::setup(...)`](https://github.com/godotengine/godot/blob/437041a983a81f9768673dc5374f644634f940f9/platform/android/tts_android.cpp#L51).

The check for `tts_enabled` and the invocation of `TTS_Android::init()` should be done within that method, after the jni methods have been set up, so we can consolidate all the init logic in one location.

Reply:
> The check for tts_enabled and the invocation of TTS_Android::init() should be done within that method, after the jni methods have been set up, so we can consolidate all the init logic in one location.

That won't work, project settings are loaded in `Main::setup`, so it's not loaded in the `GodotLib.initialize`, but  it should be possible to move it to the `GodotLib.setup`.

Reply:
Moved init to `GodotLib.setup` and `TTS_Android::setup`.